typing/typedtree.mli

```diff
@@ -27,7 +27,10 @@ module Uid = Shape.Uid
 (* We define a new constant type that can represent unboxed values.
    This is currently used only in [Typedtree], but the long term goal
    is to share this definition with [Lambda] and completely replace the
-   usage of [Asttypes.constant] *)
+   usage of [Asttypes.constant].
+   For the soundness for the uniqueness extension it is important that all
```

Comment:
I don't have a great sense for how this compares to other similar kinds of things in the compiler, but I'd be worried that this comment isn't scary enough. "Btw you might cause miscompilation if you ignore this comment". And it's easy to miss in review, when someone adds a new constant, if the diff doesn't include it.

It's better to break the build when someone needs to update something. For example, perhaps some file somewhere could have a test like:
```ocaml
type crosses_uniqueness = I_remembered_to_make_this_cross_uniqueness

let _all_constants_mode_cross_uniqueness : Typedtree.constant -> crosses_uniqueness = function
  | Const_int _ -> I_remembered_to_make_this_cross_uniqueness
  | Const_char _ -> I_remembered_to_make_this_cross_uniqueness
  (* ... *)
  | Const_unboxed_nativeint _ -> I_remembered_to_make_this_cross_uniqueness
;;
```

Ideally there's already some kind of convention for this sort of thing the compiler.